Shopify/shopify-cli
# [Scripts] Move title and descriptions from script config to project config

### WHY are these changes introduced?

CLI part of fixing https://github.com/Shopify/script-service/issues/4247



### WHAT is this pull request doing?

This PR move the "source-of-truth" for a script's name from the configuration definition (`script.config.yml / script.json`) to the cli project config (`.shopify-cli.yml`)

It also renames `script_name` to `title` for consistency. I probably should have done this in a separate PR to make it easier to review the functional changes.

### How does it do this?
- [x] Create script no longer does an update to the starter code. 
- [x] Instead the title+description (TD) are written directly by the ScriptProjectRepo during create.
- [x] The TD are passed into the PushPackage directly from the ScriptProject
- [x] The PushPackage TD is sent to Script Service.

### Important things to keep in mind during review
- The Script Project should now contain the title and description
- A Script Config should not have a title or description. This affects things like the ProjectCreators, who used to perform updates after the example code was checked out.

### How to test your changes?

- Create script and push it
- Find it in partners
- Edit the `title` field in `.shopify-cli.yml`
- Push the script again
- See the updated title in partners

### Shipping Order
- Shipping this will cause existing scripts to break, since the CLI will start looking for these values in existing `.shopify-cli.yml` files that don't have the fields.
  - [x] Mitigation: have an error message that details which fields are missing, and directs the user to adding them.
```Couldn't push script to app. Your .shopify-cli.yml is formatted incorrectly. It's missing values for extension_point_type or title. Add these values.```
- The example-repo scripts don't need title/description anymore, so they should be removed.
  - But doing so would break older versions of the CLI, which expect to need to perform the rename work
  - [ ] Mitigation: ensure that the CLI change is marked as breaking and prompts the users to update.

### Post-release steps



### Update checklist
